libpod/runtime_pod_infra_linux.go

```diff
@@ -145,6 +146,34 @@ func (r *Runtime) makeInfraContainer(ctx context.Context, p *Pod, imgName, rawIm
 		if len(p.config.InfraContainer.ExitCommand) > 0 {
 			options = append(options, WithExitCommand(p.config.InfraContainer.ExitCommand))
 		}
+		if p.config.Pid.NSMode != "private" && p.config.Pid.NSMode != "" {
+			path := p.config.Pid.Value
+			ns := spec.LinuxNamespace{}
+			pidVal := p.config.Pid.NSMode
+			if pidVal != "" {
+				if pidVal == "container" {
+					err := errors.New("Invalid Argument")
+					return nil, errors.Wrap(err, "Cannot take container in a different NS as an argument")
+				}
+				if pidVal == "ns" {
```

Comment:
What is this if statement doing? The PID namespace and PID file are not related.